pkg/endpoint/policy.go

```diff
@@ -693,11 +693,7 @@ func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 	e.BuildMutex.Lock()
 	defer e.BuildMutex.Unlock()
 
-	if err = e.RLockAlive(); err != nil {
```

Comment:
Actually, we need to check whether the endpoint is alive after taking the BuildMutex. We don't want to build dead endpoints.